client/asset/eth/eth.go

```diff
@@ -1451,13 +1475,45 @@ func (w *assetWallet) swapGas(n int, ver uint32) (oneSwap, nSwap uint64, approve
 	// If we've approved the contract to transfer, we can get a live
 	// estimate to double check.
 
+	// The amount we can estimate and ultimately the amount we can use in a
+	// single transaction is limited by the block gas limit or the tx gas
+	// limit. Estimate the number of swaps we can safely estimate here.
+	nMax := n
+	var nRemain, nFull int
+	if n > 1 && nSwap > perTxGasLimit {
+		nMax = 1
+		gas := perTxGasLimit - g.Swap
+		nMax += int(gas / g.SwapAdd)
```

Comment:
Shouldn't we use `nMax = w.maxSwapsInTx` since that's what **core** is going to do.

Reply:
I'm not sure. This is versioned...

Reply:
Why isn't `maxSwapGas` the lowest of all `gases.Swap`?

Reply:
Oh I see, the highest `maxSwapGas` creates the lowest `maxSwapsInTx`

Yeah I guess it should.